drivers/usb/usbxhci/roothub.c

```diff
@@ -0,0 +1,202 @@
+/*
+ * PROJECT:     ReactOS USB xHCI Miniport Driver
+ * LICENSE:     GPL-2.0+ (https://spdx.org/licenses/GPL-2.0+)
+ * PURPOSE:     USBXHCI root hub functions
+ * COPYRIGHT:   Copyright 2023 Ian Marco Moffett <ian@vegaa.systems>
+ */
+
+#include "usbxhci.h"
+
+#define NDEBUG
+#include <debug.h>
+
+MPSTATUS
+NTAPI
+XHCI_RH_ChirpRootPort(
+    _In_ PVOID XhciExtension,
+    _In_ USHORT Port)
+{
+    DPRINT("XHCI_RH_ChirpRootPort: UNIMPLEMENTED. FIXME\n");
+    return MP_STATUS_SUCCESS;
+}
+
+VOID
+NTAPI
+XHCI_RH_GetRootHubData(
+    _In_ PVOID XhciExtension,
+    _In_ PVOID RootHubData)
+{
+    PUSBPORT_ROOT_HUB_DATA RH_Data;
+    PXHCI_EXTENSION XhciExt;
+
+    RH_Data = (PUSBPORT_ROOT_HUB_DATA)RootHubData;
```

Comment:
This one (and the extension below) is a case of "cast from void pointer is unnecessary", though ;)

Reply:
> This one (and the extension below) is a case of "cast from void pointer is unnecessary", though ;)

Honestly that is quite silly of me. Will correct that :+1: